[e2e] e2e container fix ssh cmd for linux platform

adrianriobo opened the pull request:
this is a following PR for #3347 where ssh command tried to emulate a local session with `su`, that approach is working fine on darwin platform but is not working for linux. This PR will use that approach only for darwin and leave windows and linux as they were before PR #3347

praveenkumar commented on images/build-e2e/entrypoint.sh:
```diff
@@ -114,11 +114,16 @@ fi
+    OPTIONS+="--godog.tags=\\\"${TAGS}\\\" --godog.format=junit "
+    BINARY_EXEC+="sudo su - ${TARGET_HOST_USERNAME} -c \"PATH=$PATH:/usr/local/bin cd ${EXECUTION_FOLDER}/bin && ./${BINARY} ${OPTIONS} > ${RESULTS_FILE}.results\""
 else
-    BINARY_EXEC+="sudo su - ${TARGET_HOST_USERNAME} -c \"cd ${EXECUTION_FOLDER}/bin && ./${BINARY} ${OPTIONS} > ${RESULTS_FILE}.results\""
```
When this fails what is the error you are getting?

adrianriobo replied:
On linux I think the issue is related with systemd user services and how session is managed with `su`

adrianriobo replied:
on darwin I still would not be able to find the exact setup which is causing this requirement. On m1 there is no security profile or param which is shrinking the remote session..and the only way I found out to overcome it is using the `su` to emulate a local session but again depending on macos version default bash is different so again envs are not set the same :(, this is why I have to force the `/url/local/bin` as part of $PATH

adrianriobo commented:
/hold

adrianriobo committed:
[e2e] fix ssh cmd on e2e container

this is a following PR for [#3347](https://github.com/crc-org/crc/pull/3347) where ssh command tried to emulate a local session with su, that approach is working fine on darwin platform but is not working for linux. This PR will use that approach only for darwin and leave windows and linux as they were before the PR #3347

adrianriobo commented:
/unhold
